src/server/internal_subscription.cpp

```diff
@@ -35,6 +35,13 @@ void InternalSubscription::Stop()
 {
   DeleteAllMonitoredItems();
   Timer.cancel();
+  {
+    boost::shared_lock<boost::shared_mutex> lock(DbMutex);
+    // after Stop() the callback is useless or - as Stop is part
+    // of the shutdown sequence of OpcTcpConnection - points to
+    // a no longer existing instance of OpcTcpMessages.
+    Callback = NULL;
+  }
```

Comment:
It takes a lock of ProcessMutex in OpcTcpMessages::ForwardPublishResponse() - which may be ok, because it is (as far as I can see) only used in the boost::asio event loop. Ok - so why ProcessMutex exists at all? I'm not sure.